### DISCUSSION TITLE
Controlled Editing Mode

### DISCUSSION THREAD

**Discussion by matthewoestreich** [material-table-core/core#163] [Ideas]
From @peacechen 

> I have another feature in the pipeline : controlled editing mode. That allows the parent to put the table in edit mode instead of requiring the user to press the edit button. It would be preferable for that to be in a separate PR. When will this one be merged so I can submit the next one?

**Comment by matthewoestreich** [comment]
@peacechen isn't that already possible, or am I missing something?

```jsx
function EditableRow(props) {
  const [isEditing, setIsEditing] = useState(false);

  return (
    <div>
      <p>I am the parent</p>
      <button onClick={(e) => setIsEditing(!isEditing)}>
        {isEditing ? 'Disable' : 'Enable'} Editing
      </button>
      <MaterialTable
        editable={{
          isEditable: () => isEditing,
          onRowUpdate: (newData, oldData) =>
            new Promise((resolve, reject) => setTimeout(() => resolve(), 1000))
        }}
        data={[
          { name: 'jack', id: 1 },
          { name: 'nancy', id: 2 }
        ]}
        columns={[
          { field: 'name', title: 'Name' },
          { field: 'id', title: 'Identifier' }
        ]}
      />
    </div>
  );
}
```

**Reply by peacechen** [reply]
> @peacechen isn't that already possible, or am I missing something?

The goal is to allow the parent to programmatically enter edit mode. Currently the user must click on the edit icon (e.g. the pencil icon). The code that you provided adds the edit icon to each row, but the user must click on the icon. The same is true for bulk edit mode.

**Reply by matthewoestreich** [reply]
Ok cool I understand what you are saying now. I wasn't sure if you meant activating the edit component (eg. the editable row that shows up after you click the pencil) or if you meant enabling/disabling that pencil icon.

I really like the idea!

Are you planning on "all or nothing" (meaning, either you can edit all rows, or no rows) or will we be able to specifically target a certain row to put in edit mode?

**Reply by peacechen** [reply]
I was planning to implement it for bulk edit mode. It may not be useful for an external caller to enable editing on one specific row, but there may well be a use case that I'm not accounting for. I'll think about this some more. In the meantime there are a number of other bugfixes that need my attention.
